### ISSUE TITLE
The difference and error in tensorflow-1.8.0

### ISSUE THREAD

**Issue by LeslieZhoa** [tensorflow/tensorflow#25998]
When I use tensorflow-1.8.0, the code is different and wrong. It is shape[axis] = -1.

**Comment by ymodak** [comment]
Please provide a minimal code snippet to reproduce the issue reported here. Also provide following details: Thanks!
**System information**
- Have I written custom code (as opposed to using a stock example script provided in TensorFlow):
- OS Platform and Distribution (e.g., Linux Ubuntu 16.04):
- Mobile device (e.g. iPhone 8, Pixel 2, Samsung Galaxy) if the issue happens on mobile device:
- TensorFlow installed from (source or binary):
- TensorFlow version (use command below):
- Python version:
- Bazel version (if compiling from source):
- GCC/Compiler version (if compiling from source):
- CUDA/cuDNN version:
- GPU model and memory:

**Comment by LeslieZhoa** [comment]
1. - I have not written custom code
2. - OS Platform is Windows10
3. - Tensorflow installed by conda install --channel https://conda.anaconda.org/anaconda tensorflow-gpu=1.8.0
4. - Tensorflow version is tensorflow-gpu-1.8.0
5. - python version is 3.6.5
6. - The code of maxout is difference in https://github.com/tensorflow/tensorflow/blob/master/tensorflow/contrib/layers/python/layers/layers.py and the tensorflow-1.8.0 .
![error](https://user-images.githubusercontent.com/33117689/53227216-9dc00400-36b8-11e9-9a02-da400e4a5266.png)

my test code is as follow:

```
import tensorflow as tf
import numpy as np
layers=tf.layers
def leaky_relu(alpha):
    
    def op(inputs):
        return tf.nn.leaky_relu(inputs,alpha=alpha)
    return op
def conv_block(inputs,filters,kernel_size,strides,name,padding='valid',
               activation=leaky_relu(0.1),
               kernel_initializer=tf.contrib.layers.xavier_initializer(),
               kernel_regularizer=tf.contrib.layers.l2_regularizer(0.0004)):
    net=layers.conv2d(inputs,filters=filters,kernel_size=kernel_size,
                      activation=activation,
                      padding=padding,
                      kernel_initializer=kernel_initializer,
                      kernel_regularizer=kernel_regularizer,
                      strides=strides,name=name)
    return net
images=tf.placeholder(tf.float32,[None,64,64,3])
net=conv_block(images,128,3,2,name='conv1',activation=None)
net=tf.contrib.layers.maxout(net,64,scope='maxout1')

net=conv_block(net,256,3,2,name='conv2',activation=None)
net=tf.contrib.layers.maxout(net,128,scope='maxout2')
net=layers.max_pooling2d(net,2,2,name='pool1')

net=conv_block(net,512,3,1,name='conv3',activation=None)
net=tf.contrib.layers.maxout(net,256,scope='maxout3')`

```

the error is as follow:

```
ValueError                                Traceback (most recent call last)
<ipython-input-1-22f93208db56> in <module>()
     22 net=tf.contrib.layers.maxout(net,64,scope='maxout1')
     23 
---> 24 net=conv_block(net,256,3,2,name='conv2',activation=None)
     25 net=tf.contrib.layers.maxout(net,128,scope='maxout2')
     26 net=layers.max_pooling2d(net,2,2,name='pool1')

<ipython-input-1-22f93208db56> in conv_block(inputs, filters, kernel_size, strides, name, padding, activation, kernel_initializer, kernel_regularizer)
     16                       kernel_initializer=kernel_initializer,
     17                       kernel_regularizer=kernel_regularizer,
---> 18                       strides=strides,name=name)
     19     return net
     20 images=tf.placeholder(tf.float32,[None,64,64,3])

d:\ProgramData\Anaconda3\lib\site-packages\tensorflow\python\layers\convolutional.py in conv2d(inputs, filters, kernel_size, strides, padding, data_format, dilation_rate, activation, use_bias, kernel_initializer, bias_initializer, kernel_regularizer, bias_regularizer, activity_regularizer, kernel_constraint, bias_constraint, trainable, name, reuse)
    619       _reuse=reuse,
    620       _scope=name)
--> 621   return layer.apply(inputs)
    622 
    623 

d:\ProgramData\Anaconda3\lib\site-packages\tensorflow\python\layers\base.py in apply(self, inputs, *args, **kwargs)
    826       Output tensor(s).
    827     """
--> 828     return self.__call__(inputs, *args, **kwargs)
    829 
    830   def _add_inbound_node(self,

d:\ProgramData\Anaconda3\lib\site-packages\tensorflow\python\layers\base.py in __call__(self, inputs, *args, **kwargs)
    697           if all(hasattr(x, 'get_shape') for x in input_list):
    698             input_shapes = nest.map_structure(lambda x: x.get_shape(), inputs)
--> 699           self.build(input_shapes)
    700         try:
    701           # Note: not all sub-classes of Layer call Layer.__init__ (especially

d:\ProgramData\Anaconda3\lib\site-packages\tensorflow\python\layers\convolutional.py in build(self, input_shape)
    131       channel_axis = -1
    132     if input_shape[channel_axis].value is None:
--> 133       raise ValueError('The channel dimension of the inputs '
    134                        'should be defined. Found `None`.')
    135     input_dim = input_shape[channel_axis].value

ValueError: The channel dimension of the inputs should be defined. Found `None`.

```

**Comment by mihaimaruseac** [comment]
#22031 changed that code from 1.8.0 to the current master. Don't think it is a wrong change

**Comment by LeslieZhoa** [comment]
Hi @mihaimaruseac . I just want to show that the code in 1.8.0 is not work in my test, but it works well when I use the new code. I just want to know whether I am wrong. Thanks!

**Comment by LeslieZhoa** [comment]
Hi @mihaimaruseac . I just want to say that the code in 1.8.0 can't work well . Anyway,thanks!

**Comment by mihaimaruseac** [comment]
That's why it was fixed in #22031
